api/src/main/java/io/opencensus/stats/MeasureMap.java

```diff
@@ -51,6 +53,23 @@
    */
   public abstract MeasureMap put(MeasureLong measure, long value);
 
+  /**
+   * Associate the contextual information of an {@code Exemplar} to this {@link MeasureMap}.
+   *
+   * <p>If this method is called multiple times, the string maps should be merged. If there are
+   * multiple values with the same key in the maps, only the last value will be kept. I.e equivalent
+   * to {@code new HashMap<>().putAll(map1).putAll(map2)}.
+   *
+   * @param attachments contextual information of an {@code Exemplar}.
+   * @return this
+   * @since 0.16
+   */
+  public MeasureMap putAttachments(Map<String, String> attachments) {
+    // Provides a default no-op implementation to avoid breaking other existing sub-classes.
+    Utils.checkNotNull(attachments, "attachments");
+    return this;
+  }
+
```

Comment:
Consider alternatively 'put(String, String)' which accumulates an attachment (or 'putAttachement()'),

Reply:
Sounds good, I updated it to `putAttachement(String, String)` since we already have other methods named `put()` for `MeasureMap`.